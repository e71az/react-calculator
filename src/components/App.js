import '../style.css';
import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import 'bootstrap/dist/css/bootstrap.min.css';
// eslint-disable-next-line no-unused-vars
import calculate from '../logic/calculate';

const App = () => {
  const [allValues, setAllValues] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    const result = calculate(allValues, buttonName);

    setAllValues(result);
  };

  const { total, next } = allValues;

  return (
    <div className="calc">
      <div>
        <Display total={next || total || '0'} />
        <ButtonPanel handleClick={handleClick} />
      </div>
    </div>
  );
};

export default App;
