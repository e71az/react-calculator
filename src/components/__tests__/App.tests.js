import React from 'react';
import renderer from 'react-test-renderer';
import MainNavbar from '../NavBar';
import App from '../App';

describe('Renders App component', () => {
  it('Should render the navbar and app components correctly ', () => {
    const tree = renderer
      .create(
        <>
          <MainNavbar />
          <App />
        </>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
