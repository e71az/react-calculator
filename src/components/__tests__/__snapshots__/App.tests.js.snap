// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home component Should render the navbar and home components correctly  1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      Math Magicians
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/calc"
          href="/calc"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Calculator
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/quote"
          href="/quote"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="calc-container d-flex justify-content-between ml-5 mt-5"
  >
    <div
      className=""
    >
      <h1>
        Lets do some math!
      </h1>
    </div>
    <div
      className="calc mt-5 mr-5"
    >
      <h1
        className="result-box text-truncate text-right"
      >
        0
      </h1>
      <div
        className="button-panel"
      >
        <div
          className="group-1"
        >
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="group-2"
        >
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="group-3"
        >
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="group-4"
        >
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="group-5"
        >
          <button
            className="btn-0 rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="undefined rounded-0 calc-btn undefined undefined"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
